test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java

```diff
@@ -0,0 +1,163 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import java.net.URI;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.core.SelectedAuthScheme;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.core.interceptor.SdkInternalExecutionAttribute;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.identity.spi.Identity;
+import software.amazon.awssdk.profiles.ProfileFile;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.protocolquery.ProtocolQueryClient;
+import software.amazon.awssdk.services.protocolquery.ProtocolQueryClientBuilder;
+import software.amazon.awssdk.services.protocolquery.model.OperationWithNoInputOrOutputRequest;
+import software.amazon.awssdk.utils.CompletableFutureUtils;
+import software.amazon.awssdk.utils.StringInputStream;
+
+@WireMockTest
+public class CredentialScopeTest {
+    private ProtocolQueryClientBuilder clientBuilder;
+    private CapturingInterceptor capturingInterceptor;
+
+    @BeforeEach
+    public void setUp(WireMockRuntimeInfo wiremock) {
+        capturingInterceptor = new CapturingInterceptor();
+        clientBuilder = ProtocolQueryClient.builder()
+                                    .overrideConfiguration(o -> o.addExecutionInterceptor(capturingInterceptor))
+                                    .endpointOverride(URI.create("http://localhost:" + wiremock.getHttpPort()));
+    }
+
+    @AfterEach
+    public void reset() {
+        clearSystemProperties();
+    }
```

Comment:
Instead of clearing after, it's better to save before and restore after, in case someone is using these system properties locally for integration tests in the same run. Otherwise the integration tests may pass when run alone, but fail after this test is run.